src/util/cout_message.cpp

```diff
@@ -38,39 +35,19 @@ void console_message_handlert::print(
   if(verbosity<level)
     return;
 
-  #ifdef _WIN32
-  HANDLE out_handle=
-    GetStdHandle((level>1)?STD_OUTPUT_HANDLE:STD_ERROR_HANDLE);
-
-  // We use UTF16 when we write to the console,
-  // but we write UTF8 otherwise.
+#ifdef _WIN32
+  // We use UTF8 for unicode characters.
+  // 65001 is the magic number for the UTF8 codepage.
+  // https://docs.microsoft.com/en-us/windows/desktop/intl/code-page-identifiers
+  static bool code_page_set = false;
 
-  DWORD consoleMode;
-  if(GetConsoleMode(out_handle, &consoleMode))
+  if(!code_page_set)
   {
-    // writing to the console
-    std::wstring wide_message=widen(message);
-
-    DWORD number_written;
-
-    WriteConsoleW(
-      out_handle, wide_message.c_str(),
-      (DWORD)wide_message.size(), &number_written, NULL);
-
-    WriteConsoleW(out_handle, L"\r\n", 2, &number_written, NULL);
+    SetConsoleOutputCP(65001);
```

Comment:
Would it be possible to do this in the constructor?

Reply:
Is the `code_page_set` still necessary in this case? That's me assuming that `SetConsoleOutputCP(65001)` can safely be invoked multiple times (in the event that multiple message handlers are created).

Reply:
Yes, it's save, but apparently slow.

Reply:
Ok, then I would suggest to include this information as a comment (as otherwise it seems rather weird to maintain state in a `static` variable when the function can only ever be called in the constructor.